src/xeus_sqlite_interpreter.cpp

```diff
@@ -22,98 +23,222 @@
+    std::string is_magic(1, input.str()[0]);
+    tokenized_str.push_back(is_magic);
+
+    while(std::getline(input, segment, ' '))
+    {
+        tokenized_str.push_back(segment);
+    }
+
+
+    return tokenized_str;
+}
+
+bool interpreter::is_magic(std::vector<std::string>& tokenized_code)
+{
+    if(tokenized_code[0] == "%") {
+        tokenized_code[1].erase(0, 1);
+        std::transform(tokenized_code[1].begin(), tokenized_code[1].end(),tokenized_code[1].begin(), ::toupper);
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+
+void interpreter::load_db(std::vector<std::string> tokenized_code)
+{
+    /*
+        Loads the a database. If the open mode is not specified it defaults
+        to read and write mode.
+    */
+
+    try
+    {
+        if (tokenized_code[2] == tokenized_code.back())
+        {
+            this->db = std::make_unique<SQLite::Database>(tokenized_code[2],
+                        SQLite::OPEN_READWRITE);
+        }
+        else if (tokenized_code.back().find("rw") != std::string::npos)
+        {
+            this->db = std::make_unique<SQLite::Database>(tokenized_code[2],
+                        SQLite::OPEN_READWRITE);
+        }
+        else if (tokenized_code.back().find("r") != std::string::npos)
+        {
+            this->db = std::make_unique<SQLite::Database>(tokenized_code[2],
+                        SQLite::OPEN_READONLY);
+        }
+        //TODO: treat the case where the user doesn't input anything
+    }
+    catch(const std::runtime_error& e)
     {
+        std::cout << e.what() << std::endl;
     }
+}
+
+void interpreter::create_db(std::vector<std::string> tokenized_code)
+{
+    /*
+        Creates the a database in read and write mode.
+    */
 
-    nl::json interpreter::execute_request_impl(int execution_counter,
-                                               const std::string& code,
-                                               bool /*silent*/,
-                                               bool /*store_history*/,
-                                               nl::json /*user_expressions*/,
-                                               bool /*allow_stdin*/)
+    try
     {
-        nl::json pub_data;
-        std::string result = "";
-        std::string path_to_db = "/Users/mariana/Development/jupyter-ecosystem/xeus-sqlite/chinook.db";
-        SQLite::Database db(path_to_db, SQLite::OPEN_READWRITE|SQLite::OPEN_CREATE);
+        this->db = std::make_unique<SQLite::Database>(tokenized_code[2], SQLite::OPEN_READWRITE|SQLite::OPEN_CREATE);
+    }
+    catch(const std::runtime_error& e)
+    {
+        std::cout << e.what() << std::endl;
+    }
+}
 
-        SQLite::Statement query(db, code);
+void interpreter::parse_code(std::vector<std::string> tokenized_code)
+{
+    //TODO: maybe add a rename? https://sqlite.org/lang_altertable.html#altertabrename
+
+    std::cout << "You're using magic. " << std::endl;
+
+    if (tokenized_code[1] == "LOAD")
```

Comment:
looks good to me! Is it possible to use a switch statement with strings?

Reply:
I wanted to do that to make it pretty, but it's a lot of complicated code. I could do it! -but it'd add unnecessary layers of complication to do a simple thing just so I can use a switch, I thought it was better to use ifs, even though they're uglier, they're more straightforward. 
Thanks for the review <3

Reply:
yeah i just googled, looks like you're right. C++ is such a bad language, hehe

Reply:
ikr :p
was surprised too